drools-wb-screens/drools-wb-guided-dtree-editor/drools-wb-guided-dtree-editor-client/src/main/java/org/drools/workbench/screens/guided/dtree/client/widget/popups/ActionFieldValueEditor.java

```diff
@@ -84,85 +79,82 @@
-                                                if ( !fieldName.equals( DataType.TYPE_THIS ) ) {
-                                                    fieldListBox.addItem( fieldName );
-                                                    if ( afv.getFieldName().equals( fieldName ) ) {
-                                                        selectedIndex = fieldListBox.getItemCount() - 1;
-                                                    }
-                                                }
-
-                                            }
-
-                                            fieldListBox.addChangeHandler( new ChangeHandler() {
-                                                @Override
-                                                public void onChange( final ChangeEvent event ) {
-                                                    final String fieldName = fieldListBox.getItemText( fieldListBox.getSelectedIndex() );
-                                                    afv.setFieldName( fieldName );
-                                                    afv.setValue( null );
-                                                    initialiseValue();
-                                                }
-                                            } );
-
-                                            fieldListBox.setSelectedIndex( selectedIndex );
-                                            if ( selectedIndex == -1 ) {
-                                                fieldListBox.setSelectedIndex( 0 );
-                                                final String fieldName = fieldListBox.getItemText( 0 );
-                                                afv.setFieldName( fieldName );
-                                            }
-                                            initialiseValue();
-                                        }
-                                    } );
+        fieldListBox.setEnabled(false);
+
+        oracle.getFieldCompletions(className,
+                                   (modelFields) -> {
+                                       fieldListBox.setEnabled(modelFields.length > 0);
+                                       if (modelFields.length == 0) {
+                                           fieldListBox.addItem(GuidedDecisionTreeConstants.INSTANCE.noFields());
+                                           return;
+                                       }
+
+                                       //Add them to the ListBox
+                                       int selectedIndex = -1;
+                                       for (ModelField modelField : modelFields) {
+                                           final String fieldName = modelField.getName();
+                                           if (!fieldName.equals(DataType.TYPE_THIS)) {
+                                               fieldListBox.addItem(fieldName);
+                                               if (fieldName.equals(afv.getFieldName())) {
```

Comment:
This is the only real change. It used to be ```afv.getFieldName().equals( fieldName )``` but ```afv.getFieldName()``` might have been ```null```. I've reversed the comparison so the LHS is always a known non-```null``` value.

Reply:
Hehe, answer for question I wanted to put into review comment.